### ISSUE TITLE
Last bucket is too wide

### ISSUE THREAD

**Issue by capsuleman** [ggalmazor/lt_downsampling_java8#1]
Hello @ggalmazor!

Thanks for your implementation of LTTB algorithm, I am currently using it on a project using large timeseries!
In a recent development, I had to downsample a new type of data, and some of them have a size with the same order of magnitude than the number of buckets: ~5500 pts for 1000 buckets for example.
Because the last bucket is taking all the remaining space, it is too wide:
```
int regular_bucket_size = data.size() / numberOfBucket // 5
int last_bucket_size = data.size() / numberOfBucket + data.size() % numberOfBucket // 505 -> ~10% of data
```
Which give results like this one.
![image](https://user-images.githubusercontent.com/34281913/196490789-39c10ced-2cbd-4c88-aa88-6988f2ca18c9.png)

In this implementation in Python, bucket size are more equally split: 
https://git.sr.ht/~javiljoen/lttb-numpy/tree/master/item/src/lttb/lttb.py#L81 (cf: https://numpy.org/doc/stable/reference/generated/numpy.array_split.html)

I would be happy to contribute with a fix soon.

Best regards,
Guillaume Vagner

**Comment by ggalmazor** [comment]
Hi @capsuleman!!
Your feedback made my day! Thanks!! :) I'm super happy to know someone is using this :)

I'm also looking forward your contribution with the fix for the last bucket's size. Looking forward to reviewing it

Thanks again!

**Comment by capsuleman** [comment]
Hi @ggalmazor!
I was busy with other matters, but here is my little stone to the project!

Tell me if there is any problem

**Comment by ggalmazor** [comment]
Thanks a lot @capsuleman! I will review it asap ❤️

**Comment by ggalmazor** [comment]
FIxed by #2

**Comment by ggalmazor** [comment]
Hey, @capsuleman! I merged your PR and took the chance to do a bit of housekeeping on the repo.

I'd love to see how your contribution improves it for your use case. Can you post a screenshot maybe so that we can compare it with the one you initially posted?

I also added some examples to the [README.md](https://github.com/ggalmazor/lt_downsampling_java8/blob/main/README.md#example) file
